cmd/paasta/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+// map[string]bool is emulating a set
+func listPaastaCommands() (map[string]bool, error) {
+	cmd := exec.Command("/bin/bash", "-p", "-c", "compgen -A command paasta-")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(out.Bytes()))
+	cmds := map[string]bool{}
+	for scanner.Scan() {
+		cmds[scanner.Text()] = true
+	}
+	return cmds, nil
+}
+
+func main() {
+	var cmds map[string]bool = nil
+	var cmdPath string
+	var args []string
+
+	if len(os.Args) > 1 {
+		var err error
+		cmds, err = listPaastaCommands()
```

Comment:
nit: how about `cmds, err := listPaastaCommands()` and remove the declarations.

Reply:
can't, because `:=` means `cmds` will be local to the nearest `if {}`

Reply:
ah, right.